### PR TITLE
Simple addition to add the missing [MODEL] placeholder to the Link functionality

### PR DESCRIPTION
**Pull request by Mirkk**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: src/Link.php [hunk 1]
```diff
@@ -312,6 +312,46 @@ public static function generateLinkContents($link, CommonDBTM $item/*, bool $saf
...
             $link = str_replace("[SERIAL]", $item->getField('serial'), $link);
         }
+	if (
+	    strstr($link,"[MODEL]")
+	    && $item->isField('computermodels_id')
+	) {
+            $link = str_replace("[MODEL]",
+		    str_replace(" ", "_", Dropdown::getDropdownName("glpi_computermodels", $item->getField('computermodels_id'))),
+		    $link);
+	}
```
Comment (cedric-anne) [inline src/Link.php:322]: Code could be simplfied a lot and made generic.
```php
        $model_class = $item->getModelClass()
	if (
	    strstr($link,"[MODEL]")
	    && $model_class !== null
	) {
            $link = str_replace("[MODEL]",
		    str_replace(" ", "_", Dropdown::getDropdownName($model_class::getTable(), $item->getField($model_class::getForeignKeyField()))),
		    $link);
	}
```

File: src/Link.php [hunk 2]
```diff
@@ -312,6 +312,46 @@ public static function generateLinkContents($link, CommonDBTM $item/*, bool $saf
         ) {
             $link = str_replace("[SERIAL]", $item->getField('serial'), $link);
         }
+	if (
+	    strstr($link,"[MODEL]")
+	    && $item->isField('computermodels_id')
+	) {
+            $link = str_replace("[MODEL]",
+		    str_replace(" ", "_", Dropdown::getDropdownName("glpi_computermodels", $item->getField('computermodels_id'))),
```
Comment (cedric-anne) [inline src/Link.php:320]: What is the purpose of this `str_replace(" ", "_", ...)` ?

Comment (Mirkk) [reply src/Link.php:320]: Thanks Cedric, this str_replace(" ",_) can be removed. It is how a client preferred the handling of spaces, but I recognize now that it is not general.

Comment (cedric-anne) [reply src/Link.php:320]: You should indeed remove it.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
@Mirkk do you have time to finish the work on this PR?

**Comment by Mirkk** [comment]
Hi @cedric-anne sorry I missed this. I now changed the PR to the much more general version you suggested.

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Comment by cedric-anne** [comment]
I cherry-picked your commit and updated the existing test case. See https://github.com/glpi-project/glpi/commit/0f845cc11a313fc7eba0f2176648c6ce3a8710a5

**Pull request closed without merging**